packages/dev/core/src/Maths/tensor.ts

```diff
@@ -22,7 +22,7 @@ export type TensorLike<T> = T extends Tensor<TensorValue, infer I> ? I : never;
  * Describes a mathematical tensor.
  * @see https://wikipedia.org/wiki/Tensor
  */
-export interface Tensor<V extends TensorValue, I> {
+export interface Tensor<V extends TensorValue, I, J = I> {

```

Comment:
Introduce new generic parameters just for serialization seems to make the code too verbose

Reply:
Perhaps, but this is due to `I` for `Vector3`, `Quaternion`, and `Matrix` being the class not the interface. While I would be ok with changing `I` to `I*Like` for those classes, that would mean we use the getters (`x`/`y`/`z`) instead of the internal properties. That would result in a performance hit, which I know the BJS team is against. It defaults to `I`, so it is really only used in the 3 classes I mentioned above.